include/filewriter/filewriter_impl.hpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file filewriter_impl.hpp
+ * @author Jeffin Sam
+ */
+#ifndef MLBOARD_FILE_WRITER_IMPL_HPP
+#define MLBOARD_FILE_WRITER_IMPL_HPP
+
+#include "filewriter.hpp"
+
+namespace mlboard {
+
+
+fileWriter::fileWriter(std::string logdir,
+                       int maxQueueSize,
+                        std::size_t flushsec)
+{
+  const auto p1 = std::chrono::system_clock::now();
+  std::string currentTime =
+      std::to_string(std::chrono::duration_cast<std::chrono::seconds>(
+      p1.time_since_epoch()).count());
+  this->logdir = logdir + "/events.out.tfevents." + currentTime + ".v2";
+  close_ = true;
+  this->flushsec = flushsec;
+  size_t &maxSize = this->q.MaxSize();
+  maxSize = maxQueueSize;
+  thread_ = new std::thread(&fileWriter::writeSummary, this);
+  nexttime =
+      std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+  outfile.open(this->logdir, std::fstream::out |
+      std::ios::trunc | std::ios::binary);
+}
+
+void fileWriter::writeSummary()
+{
+  // this is a thread that will continously write summary one by one into file
+  while (true)
+  {
+    //  Break the loop if eveything is done
+    if (!(close_ || q.size() != 0)) break;
+    std::time_t timenow =
+        std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    if (timenow >= nexttime)
+    {
+      // wait for queue only if queue has any element or else it will
+      // forever
+      if (q.size() == 0) continue;
+      mlboard::Event event = q.pop();
```

Comment:
Yes , I am not sure what is the correct one pop element after flush sec or popping out everything between flush sec

Yes I think we should pop out multiple elements because it is wasting time, suppose flush sec is 4 sec and write took only 1 sec then it is wasting computer time for 3 sec wasting in the sense potential wasting is less but still 

I will change it

Reply:
https://github.com/tensorflow/tensorboard/blob/master/tensorboard/summary/writer/event_file_writer.py
I went through the Tensorflow's implementation it looks like they create a temp file in some file system and during flush they flush it out to the filesystem the entire content in the tempfile. 
In some file systems, they directly happened while writing itself so flushing would not have no effect.

Reply:
I don't think so we need two different implementation. based on file system, Let me know how should we pan out the behaviour  here ?

I think so they did because of some limitation of python not sure

Reply:
Let's keep it simple. Flush will write all elements present at that time in the queue.